cmd/infra/powervs/destroy.go

```diff
@@ -741,3 +771,196 @@ func destroyVpcLB(ctx context.Context, options *DestroyInfraOptions, subnetID st
 
 	return pagingHelper(f)
 }
+
+// createCOSClient creates COS client to interact with the COS for clean up
+func createCOSClient(serviceInstanceCRN string, cosLocation string) (*s3.S3, error) {
+	var serviceEndpoint string
+	if customEp := os.Getenv(customEpEnvNameMapping[cosService]); customEp != "" {
+		serviceEndpoint = fmt.Sprintf("s3.%s.%s.cloud-object-storage.appdomain.cloud", customEp, cosLocation)
+	} else {
+		serviceEndpoint = fmt.Sprintf("s3.%s.cloud-object-storage.appdomain.cloud", cosLocation)
+	}
```

Comment:
I see that there is `getCustomEndpointUrl` also written, can we use this by any chance? and also add more comments here and also in the doc with example of these custom endpoint how user can override.. its not very clear how to override these endpoints in the doc at the moment.

Reply:
Since it's specific to cos alone and can be reused for others thought of doing it here. Extended the `getCustomEndpointUrl ` to substitute for s3 url as well. 
Added this env name in doc custom endpoint section.

Ptal!